test/cppsuite/test_harness/runtime_monitor.h

```diff
@@ -182,14 +193,78 @@ class db_size_statistic : public statistic {
     int64_t _limit;
 };
 
+class postrun_statistic_check {
+    public:
+    explicit postrun_statistic_check(configuration *config)
+    {
+        const auto config_stats = config->get_list(POSTRUN_STATISTICS);
+        /*
+         * Each stat in the configuration is a colon separated list in the following format:
+         * - Stat name
+         * - Min limit
+         * - Max limit
+         */
+        for (const auto &c : config_stats) {
+            auto stat = split_string(c, ':');
+            if (stat.size() != 3)
+                testutil_die(-1,
+                  "runtime_monitor: Each postrun statistic must follow the format of "
+                  "\"stat_name:min_limit:max_limit\". Invalid format \"%s\" provided.",
+                  c.c_str());
+            const int min_limit = std::stoi(stat.at(1)), max_limit = std::stoi(stat.at(2));
+            if (min_limit > max_limit)
+                testutil_die(-1,
+                  "runtime_monitor: The min limit of each postrun statistic must be less than or "
+                  "equal to its max limit. Config=\"%s\" Min=%ld Max=%ld",
+                  c.c_str(), min_limit, max_limit);
+            _stats.emplace_back(std::move(stat.at(0)), min_limit, max_limit);
+        }
+    }
+    virtual ~postrun_statistic_check() = default;
```

Comment:
Is it useful to declare the destructor in our case ?

Reply:
I do this in most places just to mark it as virtual. We're not subclassing `postrun_statistic_check` but if we decided to without marking the destructor as virtual, that'd be an immediate bug. So I tend to just do this by default to avoid shooting my foot later down the track.

In this case, it's fine as long as we don't subclass it so I'm ok with removing it if you'd like.

Reply:
Ok to keep it - thanks for the details.